web/app/scripts/utils/date-utils.js

```diff
@@ -0,0 +1,87 @@
+(function() {
+    'use strict';
+
+    /* ngInject */
+    function DateUtils(LanguageState) {
+        var languageMap = {
+            'ar-sa': {
+                language: 'ar',
+                calendar: 'ummalqura',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'd MM, Y',
+                    'numeric': ' - dd/mm/yyyy'
+                }
+            },
+            'en-us': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': 'M Y',
+                    'long': 'MM d, Y',
+                    'numeric': 'mm/dd/yyyy - '
+                }
+            },
+            'exclaim': {
+                language: 'en',
+                calendar: 'gregorian',
+                formats: {
+                    'short': '\'!\'M Y',
+                    'long': '\'!\'MM d, Y',
+                    'numeric': '\'!\'mm/dd/yyyy - '
+                }
+            }
+        };
+
+        var module = {
+            getLocalizedDateString: getLocalizedDateString
+        };
+        return module;
+
+        function getLocalizedDateString(date, format, includeTime) {
+            var language = LanguageState.getSelected().id;
+            if (!language) {
+                language = 'exclaim';
+            }
+            var isRtl = LanguageState.getSelected().rtl;
+            var conversion = languageMap[language];
+            var parsedDate = $.calendars.instance()
+                .parseDate('mm/dd/yyyyy', date.toLocaleDateString()).toJSDate();
+            var convertedDate = $.calendars.instance(conversion.calendar, conversion.language)
+                .fromJSDate(parsedDate);
+            var datestring = convertedDate.formatDate(conversion.formats[format], convertedDate);
+            if(includeTime) {
+                var minutes = date.getMinutes() + '';
+                if (minutes.length < 2) {
+                    minutes = '0' + minutes;
+                }
+                var hours = date.getHours() + '';
+                if (hours.length < 2) {
+                    hours = '0' + hours;
+                }
+                var seconds = date.getSeconds() + '';
+                if (seconds.length < 2) {
+                    seconds = '0' + seconds;
+                }
+                if (isRtl && format !== 'long') {
+                    datestring = hours + ':' + minutes + ':' + seconds + '  ' + datestring;
+                } else {
+                    datestring = datestring + '  ' + hours + ':' + minutes + ':' + seconds;
+                }
+            }
+            return datestring;
+        }
+    }
+
+    /* ngInject */
+    function LocalizeDateFilter(DateUtils) {
+        function module(d, format, includeTime) {
+            return DateUtils.getLocalizedDateString(new Date(d), format, includeTime === 'time');
```

Comment:
We've had problems with the `Date()` constructor on HunchLab failing on older browsers. It might be a good idea to use a shim: https://github.com/csnover/js-iso8601/tree/lax